File: Structural/Adapter/Cpp/MediaAdapter.cpp
// MediaAdapter.cpp
#include "MediaAdapter.h"

MediaAdapter::MediaAdapter(const std::string& audioType) {
  if (audioType == "MP4") {
    advancedMediaPlayer = std::make_unique<MP4Player>();
  } else if (audioType == "VLC") {
    advancedMediaPlayer = std::make_unique<VLCPlayer>();
  }
}

void MediaAdapter::play(const std::string& audioType, const std::string& fileName) {
  if (audioType == "MP4") {
    advancedMediaPlayer->playMP4(fileName);
  } else if (audioType == "VLC") {
    advancedMediaPlayer->playVLC(fileName);
  }
}

File: Structural/Adapter/CSharp/MediaAdapter.cs
// MediaAdapter.cs
using System;

public class MediaAdapter : IMediaPlayer {
  private readonly IAdvancedMediaPlayer advancedMediaPlayer;

  public MediaAdapter(string audioType) {
    if (audioType.Equals("MP4", StringComparison.OrdinalIgnoreCase)) {
      advancedMediaPlayer = new MP4Player();
    } else if (audioType.Equals("VLC", StringComparison.OrdinalIgnoreCase)) {
      advancedMediaPlayer = new VLCPlayer();
    } else {
      throw new ArgumentException("Unsupported media type: " + audioType);
    }
  }

  public void Play(string audioType, string fileName) {
    if (audioType.Equals("MP4", StringComparison.OrdinalIgnoreCase)) {
      advancedMediaPlayer.PlayMP4(fileName);
    } else if (audioType.Equals("VLC", StringComparison.OrdinalIgnoreCase)) {
      advancedMediaPlayer.PlayVLC(fileName);
    }
  }
}
